src/arrow.rs

```diff
@@ -0,0 +1,331 @@
+        .with_primary_header(header)?
+        .with_secondary_header(&bytes)?
+        .with_user_data(user_data)?
+        .build()
+}
+
+#[cfg(test)]
+mod arrow_secondary_header_tests {
+    use super::*;
+    use crate::time::*;
+    use rand::prelude::*;
+    use uuid::uuid;
+
+    #[test]
+    fn ut_to_decode_24() -> Result<(), Error> {
+        let mut rng = thread_rng();
+
+        let resolutions = [
+            PreambleResolution::Microseconds,
+            PreambleResolution::Milliseconds,
+        ];
+
+        let epochs = [PreambleEpoch::AgencyDefined, PreambleEpoch::Jan1958];
+
+        for resolution in resolutions {
+            for epoch in epochs {
+                let tc = CdsTimeCode24::new(
+                    rng.gen_range(0..CDS_DAY_MAX_VALUE_24_BIT), // arbitrary day, obtain actual from OS
+                    rng.gen(), // arbitrary ms, obtain actual from OS
+                    resolution,
+                    epoch,
+                )?;
+
+                // arbitrary uuid, obtain actual from spacecraft
+                let uuid = uuid!("67e55044-10b1-426f-9247-bb680e5fe0c8");
+
+                // Got the header
+                let header_1 = ArrowSecondaryHeader::new(tc, uuid)?;
+                assert_eq!(header_1.time_code().preamble().epoch(), epoch);
+
+                // Empty target buffer
+                let mut bytes = [0; ARROW_SECONDARY_HEADER_LEN];
+
+                // Confirm correct number of bytes are written
+                assert_eq!(ARROW_SECONDARY_HEADER_LEN, header_1.encode(&mut bytes)?);
+
+                // Confirm the two headers are the same
+                assert_eq!(header_1, header_1);
+
+                // Create a new header from the same bytes
+                let header_2 = ArrowSecondaryHeader::decode(&bytes)?;
+                assert_eq!(header_2.time_code().preamble().epoch(), epoch);
+
+                // Confirm the two headers are the same
+                assert_eq!(header_1, header_2);
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod arrow_packet_tests {
+    use super::*;
+    use crate::time::*;
+    use rand::prelude::*;
+    use uuid::uuid;
+
+    #[test]
+    fn ut_new_packet() -> Result<(), Error> {
+        const U: usize = ARROW_DATA_SEGMENT_SIZE; // todo const generics
+
+        let mut rng = thread_rng();
+
+        // Create expected values
+        let uuid = uuid!("67e55044-10b1-426f-9247-bb680e5fe0c8"); // arbitrary
+        const USER_DATA_LEN: usize = U - ARROW_SECONDARY_HEADER_LEN;
+        let day = rng.gen_range(0..CDS_DAY_MAX_VALUE_24_BIT); // arbitrary day
+        let fine = rng.gen(); // arbitrary fine measurement
+        let apid = 0xCD;
+        let seq_count = 0xAB;
+        let pkt_type = PacketType::Telemetry;
+        let seq_flag = SequenceFlag::Unsegmented;
+        let mut user_data = [0; USER_DATA_LEN];
+        for i in 0..USER_DATA_LEN {
+            user_data[i] = rng.gen();
+        }
```

Comment:
Does the user have to fill all 28 bytes?  Is the empty portion of the array filled by 0 by default?